### ISSUE TITLE
Pandora mapping is slow

### ISSUE THREAD

**Issue by iqbal-lab** [rmcolq/pandora#347]
Placeholder as @Danderson123 keeps saying this and I don't want to lose it. Mapping reads from one sample to a 
PRG (Ecoli + a few thousand AMR genes) taking 30 mins with 64 cores. I realise our performance stats in the pandora paper are
a) way out of date
b) end to end performance of doing pandora compare on 20 samples.

**Comment by iqbal-lab** [comment]
Actually this might be a RAM/speed tradeoff - if we lazy load gene/PRGs into RAM only when we see them?

**Comment by iqbal-lab** [comment]
(Hope i'm not misrepresenting you @Danderson123 , I can delete this or update it)

**Comment by Danderson123** [comment]
This is correct, I will try to make plots of runtime vs number of reads after I have looked at the gene calling in more detail- without lazy loading the RAM usage was far higher than is reasonable for a laptop

**Comment by iqbal-lab** [comment]
I have to say @Danderson123 , first mapping to an AMR-only PRG and only keeping those reads, and then mapping just those to the big PRG, would probably make a significant difference

**Comment by iqbal-lab** [comment]
ping @Danderson123  - can i close this

**Comment by Danderson123** [comment]
Sorry, yes this can be closed!

On 8 Sep 2025, at 23:11, Zamin Iqbal ***@***.***> wrote:

﻿
[https://avatars.githubusercontent.com/u/6476513?s=20&v=4]iqbal-lab left a comment (iqbal-lab-org/pandora#347)<https://github.com/iqbal-lab-org/pandora/issues/347#issuecomment-3268190887>

ping @Danderson123<https://github.com/Danderson123> - can i close this
